src/models/statement/statement_naive.rs

```diff
@@ -51,6 +53,20 @@ impl FromMerge for StateNaive {
     }
 }
 
+impl Metadata for StateNaive {
```

Comment:
impl both `MetaData` and `StateLayout`
TODO: rename `StateLayout` into `StatementLayout`